tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -216,8 +240,23 @@ private[optimizer] abstract class OptimizerCore(
     getAncestorsOf(lhs).contains(rhs)
 
   private val isSubclassFun = isSubclass _
-  private def isSubtype(lhs: Type, rhs: Type): Boolean =
-    Types.isSubtype(lhs, rhs)(isSubclassFun)
+
+  private def isSubtype(lhs: Type, rhs: Type): Boolean = {
+    Types.isSubtype(lhs, rhs)(isSubclassFun) || {
+      (lhs, rhs) match {
+        case (LongType | ClassType(Definitions.BoxedLongClass),
+            ClassType(LongImpl.RuntimeLongClass)) =>
+          true
+
+        case (ClassType(LongImpl.RuntimeLongClass),
+            ClassType(Definitions.BoxedLongClass)) =>
+          true
+
+        case _ =>
+          false
+      }
+    }
+  }
```

Comment:
Yes. `ir/Types.scala` doesn't know about `RuntimeLong`, not should it. That is a link-time implementation detail.